buidl/taproot.py

```diff
@@ -291,7 +322,7 @@ def get_signature(self, s_sum, r, sig_hash, tweak=0):
             msg = r.bip340() + tweak_point.bip340() + sig_hash
             challenge = big_endian_to_int(hash_challenge(msg)) % N
             if tweak_point.parity:
-                s = (s_sum - challenge * tweak) % N
+                s = (-s_sum - challenge * tweak) % N
```

Comment:
What is the reason for this change? Not saying it's wrong but raises a red flag.

Reply:
Turns out there was a flaw in my logic that didn't get revealed until this line got tested (which didn't happen until the new musig2 tests)! I ran through some math and this is the correct formula.